src/plasmapy/plasma/grids.py

```diff
@@ -1107,6 +1107,51 @@ def vector_intersects(self, p1, p2):
 
         return Tmin < Tmax
 
+    def soften_edges(self, width: u.Quantity | None = None) -> None:
+        """
+        Applies a mask to soften the edges of the quantity arrays.
+
+        Ensures that quantities go to zero near the boundaries of the grid.
+
+        Parameters
+        ----------
+        width : `~astropy.units.Quantity` (optional)
+            Width of the transition region.
+            Defaults to 10% of the size of the grid.
+
+        """
+        if isinstance(width, u.Quantity):
+            width = [
+                width,
+            ] * 3
+
+        mask = np.ones(self.shape)
+        for i, pts in enumerate([self.pts0, self.pts1, self.pts2]):
+            w = 0.1 * (np.max(pts) - np.min(pts)) if width is None else width[i]
+
+            sigma = w / 4
+            pad = 2
+            x1 = np.min(pts) + pad * sigma
+            x2 = np.max(pts) - pad * sigma
+            mask *= (
```

Comment:
Mask is just two error functions going from 0-1 back to back to create a flat-top Gaussian along each axis.